api/src/main/java/org/commonjava/indy/data/StoreDataManager.java

```diff
@@ -169,6 +169,8 @@ void reload()
      */
     void asyncGroupAffectedBy( ContextualTask contextualTask );
 
+    boolean isStandaloneMode();
```

Comment:
To keep it simple, there is no need to declare this I think, we can use the config directly.

Reply:
This is going to prevent the multi class casting to InfinispanStoreDataManager in *StartupAction and *MigrationAction which make them more concise.
Coming from: https://github.com/Commonjava/indy/pull/1728#discussion_r490895029

Reply:
Oh, see. :)

Reply:
basically, the interface should only contain business method. It is better to add a property directly to specify the storemanager. e.g., now is "store.manager=ispn", and change it to "cassandra" or sth later. You may want to use a @Produce method rather than direct @Inject .

Reply:
I've defined 'store.manager.standalone=true/false' in the default-main config, and producer is also defined to be a source to be injected at runtime.
This is for StartupAction and MigrationAction in the lifecycle of Indy, it should still need to recognize the store data manager and its standalone mode, correct me if I'm wrong. @ruhan1

Reply:
Yeah, I see the producer. There is sth we can improve about it. If I understand it correctly, we will need one and only one storedatamanager. @sswguo and I think directly specify the name of store manager is better, not a boolean flag, but a string name. 
so I prefer sth like this
@producer
public StoreDataManager getStoreDataManager()
{
  if ("ispn".equals(config.getStoreManager()) or config.getStoreManager() == null )
  { 
    return new InfinispanStoreDataManager();
  }
  else {  // TODO: cassandra dbmanager
  }
}
below inject is not needed because we will have to remove it after switching to new store manager. 
    @Inject
    @StandaloneStoreDataManager
    InfinispanStoreDataManager infinispanStoreDataManager;

And we don't need to change the xxxAction's "if" condition. The instanceof is good enough because those are just infinispanStoreDataManager specific initiation code. 
I migth miss sth though. let me know if I am wrong. cc @sswguo

Reply:
btw, I am totally Okay if you think it is simple to inject a InfinispanStoreDataManager and return it in producer method. If so, I would suggest using an explicit name like @InfinispanStoreDataManager.

Reply:
I checked the code in xxxAction,  the instanceof checking is enough, once the app start up, there is only one type of storeDataManager produced, so we just need to check if that is the specific storeDataManager and do the related action then, otherwise skip the action.

Reply:
@ruhan1 @sswguo Ok, thanks for the confirmation on this, I'll tweak it.

Reply:
@ruhan1 @sswguo Another commit is pushed. For instantiation way, I'm ok to adjust it in the following cascaded debugging. This is just a sample to show the switcher. For the named annotation, the same annotation name with the class name will produce a conflict issue, so I think that would be good to specify a different one.